### ISSUE TITLE
[Bug Report]

### ISSUE THREAD

**Issue by rbavery** [aws/amazon-sagemaker-examples#3432]
please address this issue. if a sagemaker endpoint deployment hits a resource limit, it gets stuck forever and there is no option to delete it: https://stackoverflow.com/questions/65678237/sagemaker-endpoint-stuck-at-creating

**Comment by Rebecasarai** [comment]
Hi! This is happening to me right now

**Comment by chiayiffg** [comment]
I was trying to create a sagemaker endpoint using terraform CLI and I ctrl + c my command line halfway when the endpoint is being created because I found some errors in my terraform script, right now my endpoint is stucked at `Creating` status and I have no way of deleting it.

Anyone found a solution to this?

**Comment by jainalphin99** [comment]
Hi, I am facing the same issue, has anyone found a solution to this. Thanks

**Comment by chiayiffg** [comment]
hi @jainalphin99 , I went back to my AWS UI after 1 day and the status of the endpoint automatically changed from `Creating` to `Failed` , then I can delete the endpoint again, maybe try deleting it tomorrow and work with a new endpoint for now

**Comment by zionsofer** [comment]
For us it happened because of missing model or endpoint configuration, which caused it to get stuck on Creating, failing only after 1 hour!
There needs to be some way to force delete endpoints, because it can't be that misconfigurations cause a blocker for such a long time.

**Comment by david-waterworth** [comment]
I'm seeing the same thing, in the logs it keeps trying and failing to install the same package over and over

**Comment by chiayiffg** [comment]
I experienced the same with @asosnovsky-sumologic and @zionsofer, whenever I accidentally forget to install a certain package or when I misconfigured any aws role my code needs to retrieve, I will need to wait close to 20 mins for the status to become fail before I can redeploy again 

@david-waterworth From what I observe, when you first deploy the endpoint, sagemaker will try to call your /ping endpoint, even if something is failing, the /ping health check will continue for 20mins (from my experience) before the endpoint gives up and return `Failed` status

To unblock yourself, I suggest working with a new endpoint with different name first because waiting for 20mins to be able to redeploy a change again is quite time consuming and demotivating

**Comment by randallbpotter15** [comment]
Agree why can't a sig be sent to the server to terminate the running processes and send it into a failed state?
It's currently trying the same thing over and over again.

**Comment by grraffe** [comment]
+1 because it can be stuck when using Graviton2 without using arm64 docker image.

**Comment by ylhsieh** [comment]
+1 no way to interrupt creating an endpoint is really a waste of time. Feels like it is a way to get more money from users waiting for timeout to finish.

**Comment by cceyda** [comment]
+1 currently stuck at creating for 20+ minutes because it can't find pip package(due to typo). Can't stop, can't delete... the more I use sagemaker the more I'm shocked how much obvious basic functionality is missing

**Comment by em-eman** [comment]
it is quite annoying and i have to delete endpoint config in order to make it fail before 20mins

**Comment by Rhuax** [comment]
Just got in the same situation.. I think it is a critical feature to implement!!

**Comment by bjmrevilla** [comment]
would really love for this to be added. Currently stuck right now in `Creating` status. :(

**Comment by danb27** [comment]
+1

**Comment by RahulJana** [comment]
I tried creating sagemaker endpoint from the notebook instance, It stuck in that state for ~30 mins. So I gave a keyboard interrupt in the notebook cell, to stop that process. Now it is stuck at `creating` status.

**Comment by nisalupendra** [comment]
Still the same issue, couldn't agree more with @ylhsieh , probably a try at scraping that extra penny from the customer, otherwise don't see why there cannot be kill switch, this doesn't depend on anything except the model and the modelConfig which are both static resources.

**Comment by bcarsley** [comment]
Yup same here! really sucks -- I assume the timeout on the creating => failed is in part based on the parameter "container_startup_health_check_timeout" (at least in my case since I'm trying to deploy a fine-tuned LLaMA as a HuggingFacePredictor) ... still they should do something about it! been waiting here for about 45mins + smh

**Comment by bcarsley** [comment]
It does seem like deleting the model image that the deployment is based upon frees up the resources slightly faster than waiting for "Failed", as you throw an arn error by deleting the model a deployment in the "creating" status is working off of (which then enables you to delete the deployment, at least in my case)... a very risky and probably unadvisable workaround!

**Comment by davidshhh** [comment]
It doesn't seem like sagemaker is made for production systems honestly.

**Comment by orcaman** [comment]
Same issue. Can the SageMaker team fix this? It's really annoying

**Comment by MarkoMilos** [comment]
How did you solve this problem? 

I have a SageMaker Endpoint that is "In Service" with InferenceComponent that is stuck in the "Creating" state. In order to delete the endpoint (to avoid charges for the underlying instance) I need to delete the InferenceComponent - but I can't while it is in the "Creating" state - it's been like that for almost 2 days now - clearly bugged. In the meantime I'm charged almost 2USD per hour.

I can't delete it via CLI, nor via AWS Console, nor via SageMaker Studio... -.-

See below

`aws sagemaker list-endpoints`
```json
{
    "Endpoints": [
        {
            "EndpointName": "llama2-endpoint",
            "EndpointArn": "arn:aws:sagemaker:eu-central-1:592532275118:endpoint/llama2-endpoint",
            "CreationTime": "2023-12-19T16:03:34.976000+01:00",
            "LastModifiedTime": "2023-12-19T16:05:22.443000+01:00",
            "EndpointStatus": "InService"
        }
    ]
}
```

`aws sagemaker list-inference-components`

```json
{
    "InferenceComponents": [
        {
            "CreationTime": "2023-12-19T16:22:21.345000+01:00",
            "InferenceComponentArn": "arn:aws:sagemaker:eu-central-1:592532275118:inference-component/llama2-7b-20231219-152221",
            "InferenceComponentName": "llama2-7b-20231219-152221",
            "EndpointArn": "arn:aws:sagemaker:eu-central-1:592532275118:endpoint/llama2-endpoint",
            "EndpointName": "llama2-endpoint",
            "VariantName": "variant-1",
            "InferenceComponentStatus": "Creating",
            "LastModifiedTime": "2023-12-19T16:22:22.333000+01:00"
        }
    ]
}
```

If I try to delete it
`aws sagemaker delete-inference-component --inference-component-name "llama2-7b-20231219-152221"`
The error returned is
```
An error occurred (ValidationException) when calling the DeleteInferenceComponent operation: Cannot delete inference component "arn:aws:sagemaker:eu-central-1:592532275118:inference-component/llama2-7b-20231219-152221" while it is in state "CREATE_IN_PROGRESS".```

**Comment by sagar1001** [comment]
Same issue for me as well. Unable to delete it due to Status CREATE_IN_PROGRESS. Any option to set inference component status to Failed or deleted from cli?

Any solutions, please suggest.

**Comment by Seven2Nine** [comment]
> +1 no way to interrupt creating an endpoint is really a waste of time. Feels like it is a way to get more money from users waiting for timeout to finish.

you are right, A year later aws still hasn't fixed this issue

**Comment by milanshen** [comment]
Same issue here, we filed a high pri ticket but still no response after almost a day

**Comment by zdev24** [comment]
I had the similar issue:
- The inference endpoint has the status of "InService".
- There is no Model associated.
- There is no endpoint configuration associated with the "InService" endpoint. Maybe it was removed when I click delete the model.
However, it still cost me $18 after 1 day and maybe will continue costing in the next day.
I can not delete the endpoint even with CLI, the error message: "An error occurred (ValidationException) when calling the DeleteEndpoint operation: Cannot delete endpoint with inference component associated. Please delete inference component and try it again."

**Comment by MarkoMilos** [comment]
I've upgraded my AWS support plan to Premium in order to find out what is going on and how to delete the endpoint.

After a video call, I got confirmation from AWS technical support that this is indeed a known issue on their side. They do not have a clear ETA when the issue will be fixed.

Apparently, they will refund me for the generated cost but not while resources are running and the issue is not fixed (for which they do not have any ETA - which does not give me that much hope since this issue was opened in May 2022 and is still active).

Meanwhile, until they fix the issue I'm charged 50$ USD per day. So far with SageMaker I feel like a Beta tester that has to pay for testing clearly bugged service...

**Comment by zdev24** [comment]
@MarkoMilos  I created a ticket I received the guideline to delete the Domain in Sagemaker. They explained that the endpoint does not cause the cost. I need to wait for 1 more day to see if it continue cost, however, you can try.

**Comment by sagar1001** [comment]
My issue got fixed yesterday. AWS unblocked and failed the IC from their end and asked me to delete IC and end point. This time it worked, and processed the billing adjustment.

**Comment by ziadbadwy** [comment]
i faced same problem at model deployment and i customized the time from **container_startup_health_check_timeout**

llm = llm_model.deploy(
  initial_instance_count=1,
  instance_type=instance_type,
  container_startup_health_check_timeout=health_check_timeout, **# 10 minutes to be able to load the model**

**Comment by komaldedhia** [comment]
+1
It is 2024 but same issue since 2 year.

**Comment by mrseeker** [comment]
bump, issue still exists.

**Comment by ziadbadwy** [comment]
> bump, issue still exists.
it takes only i think about 30 min to 2 hr

**Comment by staghado** [comment]
still exists

**Comment by MFaiqKhan** [comment]
It is 2025 July, this bug still exist .
unable to delete the endpoint in creating phase.

**Comment by RahulJana** [comment]
This bug still exists :(

**Comment by vivasvan-patel-ev** [comment]
Same here!

**Comment by sanikakulkarniss** [comment]
Mine failed in about 27 minutes. It kept trying to reinstall dependencies till then.

**Comment by emaksOne** [comment]
I also faced this issue and I tried to delete model, delete endpoint_config and then I was able to delete endpoint itself

**Comment by lp-noel** [comment]
come onnnnn

**Comment by pedrodotavila** [comment]
it's january 2025 and still exist

**Comment by etiennebonnafoux** [comment]
This feel like a Youtube Video commentary session.

> Who is there in 2025 !!!??

**Comment by mrseeker** [comment]
> i faced same problem at model deployment and i customized the time from **container_startup_health_check_timeout**
> 
> from sagemaker.huggingface import HuggingFaceModel
> 
> llm = llm_model.deploy( initial_instance_count=1, instance_type=instance_type, container_startup_health_check_timeout=300) **# 10 minutes to be able to load the model**

For those wondering, AWS does not fix it, this is the working workaround.

**Comment by etiennebonnafoux** [comment]
Do you have the equivalent in terraform ?

I guess it should be something like

```hcl
resource "aws_sagemaker_model" "llama_model" {
  name               = "llama-hf"
  execution_role_arn = aws_iam_role.sagemaker_role.arn

  primary_container {
    image = "141502667606.dkr.ecr.eu-west-1.amazonaws.com/tei-cpu:2.0.1-tei1.2.3-cpu-py310-ubuntu22.04"
    environment = {
      HF_MODEL_ID    = "meta-llama/Llama-3.1-8B-Instruct"
      HF_API_TOKEN   = var.llama_api_token
      RUST_BACKTRACE = "full"
      container_startup_health_check_timeout_in_seconds = 300
    }
  }
}
```
Can you confirm @mrseeker ?

**Comment by Epiphero** [comment]
I don't see any way to set the `container_startup_health_check_timeout` in CDK ... anyone have a workaround that works with the CDK API?

**Comment by Epiphero** [comment]
I'm using CDK to define the model and the AWS SDK to dynamically create the endpoint.  I don't see any `container_startup_health_check_timeout_in_seconds` in either API.  Has anyone found a workaround that can shorten the time out in this scenario?

**Comment by sky87** [comment]
@Epiphero not sure if/how it's exposed in the CDK, but in CF there is this
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sagemaker-endpointconfig-productionvariant.html#cfn-sagemaker-endpointconfig-productionvariant-containerstartuphealthchecktimeoutinseconds

**Comment by hhussain01-tpr** [comment]
Still facing this issue

**Comment by mrseeker** [comment]
I think the tag #wontfix will be most relevant here.

**Comment by dannylinden** [comment]
October 2025, issue still exist

**Comment by lilac** [comment]
Still see this issue, that's annoying.

**Comment by ryan-petco** [comment]
Still here, happy 2026 all!
